docs/advanced.md

```diff
@@ -1,5 +1,32 @@
 # Advanced Usage
 
+## Handling very large integers
+
+Python's `int` type is closer in structure to C#.NET's `System.Numerics.BigInteger` than to `System.Int64`. This means that when you are working with very large integers in Python, you may need to use the `BigInteger` type in C# to handle the results.
+
+You can use this using the TypeConverter class to convert between `BigInteger` and `PyObject` instances. Here's an example of how you can call a Python function that returns a very large integer:
+
+```csharp
+using CSnakes.Runtime.Python;
+using System.ComponentModel;
+using System.Numerics;
+
+const string number = "12345678987654345678764345678987654345678765";
+TypeConverter td = TypeDescriptor.GetConverter(typeof(PyObject));
+// Something that is too big for a long (I8)
+BigInteger input = BigInteger.Parse(number);
+
+using (GIL.Acquire())
+{
+    using PyObject? pyObj = td.ConvertFrom(input) as PyObject;
+
+    // Do stuff with the integer object
+
+    // Convert back
+    BigInteger integer = (BigInteger) td.ConvertTo(pyObj, typeof(BigInteger))!;
```

Comment:
We should make a method on `PyObject` that wraps the converter so people don't have to access it themselves, maybe have `public static T? PyObject.Create<T>(T value) => td.ConvertFrom(input);` and then another method on `PyObject` that is an instance method to do the `ConvertTo` (or is that available with the `As<T>` method?).

This way we can completely hide away the converter part and if we want to change it in the future we're not going to be breaking people.